pkg/scheduler/scheduler.go

```diff
@@ -86,8 +85,41 @@ type Scheduler struct {
 	schedulerEstimatorWorker util.AsyncWorker
 }
 
+type schedulerOptions struct {
+	// enableSchedulerEstimator represents whether the accurate scheduler estimator should be enabled.
+	enableSchedulerEstimator bool
+	// schedulerEstimatorTimeout specifies the timeout period of calling the accurate scheduler estimator service.
+	schedulerEstimatorTimeout metav1.Duration
+	// schedulerEstimatorPort is the port that the accurate scheduler estimator server serves at.
+	schedulerEstimatorPort int
+}
+
+// Option configures a Scheduler
+type Option func(*schedulerOptions)
```

Comment:
Would it be a little bit confusing for this Option `function` and [struct](https://github.com/karmada-io/karmada/blob/28a6c2c7672a5efd504c9c264bf2c6f53e5e89b4/cmd/scheduler/app/options/options.go#L28)?

Reply:
They belong to different packages, looks good to me. Or change to `type Configurator func(*schedulerConfigurators)` ?

Reply:
I'm thinking about why we need to introduce the `Option`, can we reuse the legacy option? https://github.com/karmada-io/karmada/blob/28a6c2c7672a5efd504c9c264bf2c6f53e5e89b4/cmd/scheduler/app/options/options.go#L28

I know, there is a lot of `option`s in the Kubernetes scheduler, but my concern is the readability.

Reply:
it is better to describe the benefits to import the **Option**

Reply:
Yes, it's not necessary but I think meaningful. [karmada/cmd/scheduler/app/options/options.go](https://github.com/karmada-io/karmada/blob/28a6c2c7672a5efd504c9c264bf2c6f53e5e89b4/cmd/scheduler/app/options/options.go#L28) holds lots of options we don't need to instantiate scheduler, as time goes on, there will be more options introduced. It's hard to maintain as we have to check the code to see which options we need and which one is required by other components, so we should only keep the necessary ones. Also it's easy to mock the `NewScheduler` tests then.